.github/workflows/unit-test.yml

```diff
@@ -0,0 +1,43 @@
+name: Unit tests
+
+on:
+  push:
+    branches:
+      - main
+    tags:
+      - "v*.*.*"
+  pull_request:
+
+jobs:
+  test:
+    strategy:
+      matrix:
+        go-version: [1.16]
+        os: [ubuntu-latest]
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/setup-go@v3
+        with:
+          go-version: ${{ matrix.go-version }}
+      - uses: actions/checkout@v3
+      - run: go test github.com/peggyjv/sommelier/.../x/...
+
+  test-cache:
```

Comment:
What's the reasoning behind there being two different unit testing jobs (`test` and `test-cache`)? Also is this based off of the workflow that used to exist in this repo for unit testing or is this from elsewhere?

Reply:
Ah yeah I was trying to cache tests. Ill remove this one. And no this is a new workflow for unit testing I spun up after looking at how a couple other repos were doing golang unit tests. If theres another template we prefer to use I can update to that

Reply:
Updated.